File: app/src/main/java/pulkit/com/torontowastewizard/Activity/SearchActivity.java
package pulkit.com.torontowastewizard.Activity;

import android.content.Intent;
import androidx.appcompat.app.AppCompatActivity;
import android.os.Bundle;
import androidx.recyclerview.widget.DividerItemDecoration;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;
import android.view.WindowManager;

import java.util.ArrayList;

import pulkit.com.torontowastewizard.Adapter.WasteAdapter;
import pulkit.com.torontowastewizard.Model.Waste;
import pulkit.com.torontowastewizard.R;

public class SearchActivity extends AppCompatActivity {

    RecyclerView wasteRecyclerView;

    ArrayList<Waste> wasteArrayList = new ArrayList<>();

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_search);

        getWindow().setFlags(WindowManager.LayoutParams.FLAG_FULLSCREEN, WindowManager.LayoutParams.FLAG_FULLSCREEN);

        Intent intent = getIntent();
        wasteArrayList = (ArrayList<Waste>) intent.getSerializableExtra("wasteArrayList");

        wasteRecyclerView = findViewById(R.id.wasteRecyclerView);

        LinearLayoutManager layoutManager = new LinearLayoutManager(this);
        wasteRecyclerView.setLayoutManager(layoutManager);

        DividerItemDecoration itemDecor = new DividerItemDecoration(this, DividerItemDecoration.VERTICAL);
        itemDecor.setDrawable(this.getResources().getDrawable(R.drawable.line_divider));
        wasteRecyclerView.addItemDecoration(itemDecor);

        WasteAdapter adapter = new WasteAdapter(wasteArrayList);
        wasteRecyclerView.setAdapter(adapter);

//        ParkingReportAdapter adapter = new ParkingReportAdapter(wasteArrayList);
//        wasteRecyclerView.setAdapter(adapter);
    }
}

File: app/src/main/java/pulkit/com/torontowastewizard/activity/SearchActivity.kt
package pulkit.com.torontowastewizard.activity

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import androidx.recyclerview.widget.DividerItemDecoration
import androidx.recyclerview.widget.LinearLayoutManager
import android.view.WindowManager
import android.databinding.DataBindingUtil
import java.util.ArrayList
import pulkit.com.torontowastewizard.adapter.WasteAdapter
import pulkit.com.torontowastewizard.model.Waste
import pulkit.com.torontowastewizard.R
import pulkit.com.torontowastewizard.databinding.ActivitySearchBinding

class SearchActivity : AppCompatActivity() {

    companion object {
        private val TAG: String = SearchActivity::class.java.simpleName
    }

    private lateinit var binding : ActivitySearchBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        binding = DataBindingUtil.setContentView(this, R.layout.activity_search)

        window.setFlags(WindowManager.LayoutParams.FLAG_FULLSCREEN, WindowManager.LayoutParams.FLAG_FULLSCREEN)

        val wasteArrayList = intent.getSerializableExtra("wasteArrayList") as ArrayList<Waste>
        val adapter = WasteAdapter(wasteArrayList)
        val layoutManager = LinearLayoutManager(this)
        val itemDecor = DividerItemDecoration(this, DividerItemDecoration.VERTICAL)
        itemDecor.setDrawable(this.resources.getDrawable(R.drawable.line_divider))

        binding.wasteRecyclerView.apply {
            this.layoutManager = layoutManager
            addItemDecoration(itemDecor)
            this.adapter = adapter
        }
    }
}
